controllers/servicefence_controller.go

```diff
@@ -640,6 +640,31 @@ func (r *ServicefenceReconciler) newSidecar(sf *lazyloadv1alpha1.ServiceFence, e
 		},
 	}
 
+	// Fetch the Service instance
+	nsName := types.NamespacedName{
+		Name: sf.Name,
+		Namespace: sf.Namespace,
+	}
+	svc := &corev1.Service{}
+	if err := r.Client.Get(context.TODO(), nsName, svc); err != nil {
+		svc = nil
+		log.Errorf("get service %s error, %+v", nsName, err)
+		return nil, err
+	}
+
+	if sf.Spec.WorkloadSelector != nil && sf.Spec.WorkloadSelector.FromService && svc != nil  {
```

Comment:
if explicitly specified to use selector from service, we should wait service to be ready rather than fullfilling a incorrect selector item.

Reply:
if svc == nil, will return directly before this check

```go
	svc := &corev1.Service{}
	if err := r.Client.Get(context.TODO(), nsName, svc); err != nil {
		if errors.IsNotFound(err) {
			log.Warningf("cannot find service %s for servicefence, skip sidecar generating", nsName)
			return nil, nil
		} else {
			log.Errorf("get service %s error, %+v", nsName, err)
			return nil, err
		}
	}